optimum/exporters/onnx/config.py

```diff
@@ -125,6 +141,13 @@ def _create_dummy_input_generator_classes(self, **kwargs) -> List["DummyInputGen
 
         return dummy_inputs_generators
 
+    def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self._behavior is ConfigBehavior.DECODER:
+            reference_model_inputs["input_ids"] = reference_model_inputs.pop("decoder_input_ids")
+            reference_model_inputs["encoder_hidden_states"] = reference_model_inputs.pop("encoder_outputs")[0]
+            reference_model_inputs["encoder_attention_mask"] = reference_model_inputs.pop("attention_mask")
+        return reference_model_inputs
```

Comment:
Thinking out loud: could `torch_to_onnx_input_map` and a loop be used in there?

Reply:
I get it could, but you're right, let's keep it like that for now as it is more readable.

Reply:
All good